src/test/java/test/listeners/github1465/IssueTest.java

```diff
@@ -0,0 +1,52 @@
+package test.listeners.github1465;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.testng.ITestNGListener;
+import org.testng.TestNG;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlSuite;
+import test.SimpleBaseTest;
+
+public class IssueTest extends SimpleBaseTest {
+
+    @Test(dataProvider = "dp")
+    public void reproduceIssue(XmlSuite.FailurePolicy policy, String[] expectedMsgs) {
+        TestNG testNG = create(ExampleClassSample.class);
+        testNG.setConfigFailurePolicy(policy);
+        ExampleClassListener listener = new ExampleClassListener();
+        testNG.addListener((ITestNGListener) listener);
+        testNG.run();
+        assertThat(listener.messages.size()).isEqualTo(4);
+        String[] expected = new String[]{
```

Comment:
Or you can directly pass strings in the varargs.

Reply:
In either case, it's still going to be the same right since varargs are internally just arrays

Reply:
True! I'm just sharing my own code style in order to try to not have too many different styles. But feel free to ignore such kind of comments ;)

Reply:
@juherr - Yes I kind of realized that part. But then again, I guess it would be good if we could not force the coding style on contributors, because everyone has their own style of coding and its hard to bring consensus on it. It also elongates code reviews and turn around time and thus a PR can perhaps take a longer time. So it would be good if we can avoid/skip those comments which can be branded as coding style.

Reply:
@krmahadevan I disagree because I think it will be more and more difficult the read the code if we have too many styles (it is already the case IMO), and trying to have a close style will help.
A consensus is not a goal and will be a waste of time, but style comments will just highlight divergences. The final decision is yours and style should not block the merge (except specific cases :p). WDYT?

Reply:
@juherr - Asking for a PR which adheres to an undocumented coding style only makes it more worse. It deteriorates the enthusiasm in a person to raise a PR. Logging comments in a PR are meant to get addressed, but logging a comment and then leaving the decision to the PR creator adds more confusion (which to fix and which not to). If coding styles (if we are looking at following a common one) are to be followed in PR, then its better to have them documented. That way there's no room for confusion and a person who like to raise a PR would be informed even before the PR is raised about the obvious set of areas wherein flags are going to be raised (if the desired conventions/styles are not followed). 

I am not against having a coding style defined for a project, I just would like to see it documented if its asked to be followed. That way it helps keep things easier and PR reviews/merges are smoother.

Reply:
ping @cbeust - Cedric, please add your thoughts as well on this, so that we can come to a consensus and start working on building the guidelines. I think it would be helpful to anyone who wants to contribute to TestNG as well, in general, if such guidelines are created.

Reply:
@krmahadevan 
> It deteriorates the enthusiasm in a person to raise a PR.

I hope it is not your case, it was not a goal.

About the guideline, I've already proposed to follow Google style on new/modified code because it is well documented and it provides IDE files (https://google.github.io/styleguide/javaguide.html / https://github.com/google/styleguide).

But here, in the specific case of array versus varargs, I don't know if we need debate: I find varargs nicer but I'm not sure it is more readable.

Reply:
I'm pretty strict on syntactic issues (e.g. indentation, naming) but less so on semantics (what you guys are discussing above).

At the end of the day, I don't think having different styles like this makes it really hard to read the code and minimizing these rules encourages people to submit a PR.

Reply:
@cbeust - Thanks for your feedback.

@juherr 

> I hope it is not your case, it was not a goal.

Never at all :) TestNG contribution has now turned into too big of an addiction and I would like to keep it that way. I get to learn a lot of stuff in the name of contributing. So I am not going to let anything hinder that.